### PR TITLE
Fix formDeposits logic

### PR DESCRIPTION
**Pull request by jameswhollister**
## Ticket
* 920bf72903f0cdad7965a633de845288909c4a3a
* 726971174b0b6397c87129946318585b314049d1
* 9864ecf1579f6c2471fe92d3bcba78b1c7a396b7

## Changes
* Fix formDeposits logic
* Improve order confirmation dialog verbiage to make more clear
* Re-Fetch holdings upon a websocket order message

### CODE REVIEW

File: src/renderer/services/dex.js [hunk 1]
```diff
@@ -590,16 +590,22 @@ export default {
...
       if (sellAssetHolding.canPull === false && order.quantity.isGreaterThan(0)) {
         // this is an MCT based token that can not be pulled from our DEX contract, have to send a deposit first
-        totalQuantityToSell = order.side === 'Buy' ? order.quantity.multipliedBy(order.price) : order.quantity;
+        depositMakerQuantity = true;
       } else if (sellAssetHolding.decimals < 8) {
         // this is a token with < 8 decimals, NEO for example, make the deposit of the minimum amount needed to make the order
-        totalQuantityToSell = order.side === 'Buy' ? order.quantity.multipliedBy(order.price) : order.quantity;
+        depositMakerQuantity = true;
+      } else if (order.offersToTake.length > 0 && order.quantityToMake.isGreaterThan(0)) {
+        // we have maker and taker quantities, need to deposit the maker quanity first because we don't know the order they will be confirmed
```
Comment (jsolman) [inline src/renderer/services/dex.js]: I think deposit would be necessary regardless of the order they are confirmed. Maybe update the comment.

File: src/renderer/components/modals/OrderConfirmationModal.vue [hunk 2]
```diff
@@ -24,7 +24,7 @@
             </span>
             <span class="price" v-else>
               {{ $formatNumber($store.state.orderToConfirm.expectedQuantityToReceive) }}
-              {{ $store.state.orderToConfirm.market.quoteCurrency }}
+              {{ $store.state.orderToConfirm.market.baseCurrency }}
```
Comment (jsolman) [inline src/renderer/components/modals/OrderConfirmationModal.vue]: Do we want this to be different based on whether the maker or the taker?

File: src/renderer/services/dex.js (hunk 1 above)
Comment (jameswhollister) [reply src/renderer/services/dex.js]: If it confirmed the takers first then the makers would pull the required tokens in during addOffer.

Comment (jsolman) [reply src/renderer/services/dex.js]: ah yes, thanks for clarifying

**Pull request merged**